homeassistant/components/openweathermap/sensor.py

```diff
@@ -313,8 +323,38 @@ def __init__(
     @property
     def native_value(self) -> StateType:
         """Return the state of the device."""
+        if self.entity_description.key == ATTR_API_NATIONAL_WEATHER_ALERTS:
```

Comment:
Could we maybe implement an `update_attr()` function instead which does the calcs when the updates comes in instead of inside the properties?
Example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/trafikverket_camera/sensor.py#L127-L139